src/core/useKTX2.tsx

```diff
@@ -0,0 +1,39 @@
+import { Texture } from 'three'
+import { useLoader, useThree } from '@react-three/fiber'
+import { useEffect } from 'react'
+import { KTX2Loader } from 'three-stdlib'
+import { IsObject } from './useTexture'
+
+export function useKTX2<Url extends string[] | string | Record<string, string>>(
+  input: Url,
+  basisPath?: string
+): Url extends any[] ? Texture[] : Url extends object ? { [key in keyof Url]: Texture } : Texture {
+  const gl = useThree((state) => state.gl)
+  const textures = useLoader(KTX2Loader, IsObject(input) ? Object.values(input) : (input as any), (loader: any) => {
+    loader.detectSupport(gl)
+    loader.setTranscoderPath(basisPath || '/basis/')
```

Comment:
```suggestion
  basisPath?: string =  '/basis/'
): Url extends any[] ? Texture[] : Url extends object ? { [key in keyof Url]: Texture } : Texture {
  const gl = useThree((state) => state.gl)
  const textures = useLoader(KTX2Loader, IsObject(input) ? Object.values(input) : (input as any), (loader: any) => {
    loader.detectSupport(gl)
    loader.setTranscoderPath(basisPath)
```

tiny change moving the default to the args instead